test/unit/models/document/paginated_history_test.rb

```diff
@@ -64,6 +45,105 @@ class PaginatedHistoryTest < ActiveSupport::TestCase
     assert_equal expected_actions, actual_actions
   end
 
+  test "#audit_trail correctly determines actors" do
+    history = Document::PaginatedHistory.new(@document, 1)
+    audit_trail = history.audit_trail
+    actual_actors = audit_trail.map(&:actor)
+    expected_actors = [@user,
+                       @user,
+                       @user,
+                       @user2,
+                       @user2,
+                       @user2,
+                       @user2,
+                       @user2,
+                       @user2,
+                       @user,
+                       @user]
+
+    assert_equal expected_actors, actual_actors
+  end
+
+  test "saving after changing the state records a state change action" do
+    @newest_edition.state = "published"
+    @newest_edition.save!
+
+    history = Document::PaginatedHistory.new(@document, 1)
+    audit_trail = history.audit_trail
+
+    assert_equal "published", audit_trail.first.action
+  end
+
+  test "saving without any changes does not get recorded as an action" do
+    history_before = Document::PaginatedHistory.new(@document, 1)
+    audit_trail_before = history_before.audit_trail
+
+    @newest_edition.save!
+    history_after = Document::PaginatedHistory.new(@document, 1)
+    audit_trail_after = history_after.audit_trail
+
+    assert_equal audit_trail_before.size, audit_trail_after.size
+  end
+
+  test "rejecting records a rejected action" do
+    @newest_edition.submit!
+    acting_as(@user2) do
+      @newest_edition.reject!
+    end
+
+    history = Document::PaginatedHistory.new(@document, 1)
+    audit_trail = history.audit_trail
+
+    assert_equal "rejected", audit_trail.first.action
+    assert_equal @user2, audit_trail.first.actor
+  end
+
+  test "saving after changing an attribute without changing the state records an update action" do
+    acting_as(@user) do
+      @newest_edition.title = "foo"
```

Comment:
Do we need to set the title here?

Reply:
I think the title change is necessary because it provides the context of the test (where an attribute is changed but not the state)

Reply:
That makes sense 👍